### PR TITLE
Warp 20231020

### PR DESCRIPTION
**Pull request by adi-wesolowski-fandom**

### CODE REVIEW

File: src/platforms/shared/bidders/bidders-state.setup.ts [hunk 1]
```diff
@@ -66,10 +66,7 @@ export class BiddersStateSetup implements DiProcess {
 				this.enableIfApplicable('testBidder', 'icPrebidTestBidder');
 			}
 
-			context.set(
-				'bidders.prebid.auctionDelay',
-				this.instantConfig.get('icPrebidAuctionDelay', 50),
-			);
+			context.set('bidders.prebid.auctionDelay', this.instantConfig.get('icPrebidAuctionDelay', 0));
```
Comment (adi-wesolowski-fandom) [inline src/platforms/shared/bidders/bidders-state.setup.ts:69]: why 0?

File: src/core/phase/browser-phase-promise.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { logger } from '../utils';
+
+function browserPhasePromise(
+	check: () => boolean,
```
Comment (adi-wesolowski-fandom) [inline src/core/phase/browser-phase-promise.ts:4]: name could be misleading. maybe "checkCallback" oraz "validateFunction"?

File: src/core/phase/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './browser-phase-promise';
```
Comment (adi-wesolowski-fandom) [inline src/core/phase/index.ts:1]: we already know that combining imports leads to unnecessary complications and larger code. So do we want to continue using this approach?

File: src/core/phase/browser-phase-promise.ts (hunk 2 above)
Comment (jbienkowski-fandom) [reply src/core/phase/browser-phase-promise.ts:4]: or `predicate`

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const DEFAULT_PARTNERS_PHASE_WAIT_TIMEOUT = 700;
+const DEFAULT_PARTNERS_PHASE_DELAY = 100;
+
+const DEFAULT_AD_CALL_WAIT_TIMEOUT = 2000;
+
+export class AdEnginePhasesSetup implements DiProcess {
+	execute(): void {
+		adEnginePhases.configuration = domContentLoadedPromise(
+			context.get('options.phases.configurationTimeout') ||
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:22]:
```suggestion
			context.get('options.phases.configurationTimeout') ??
```

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+				DEFAULT_CONFIGURATION_PHASE_WAIT_TIMEOUT,
+			context.get('options.phases.configurationDelay') || DEFAULT_CONFIGURATION_PHASE_DELAY,
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:24]:
```suggestion
			context.get('options.phases.configurationDelay') ?? DEFAULT_CONFIGURATION_PHASE_DELAY,
```

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		);
+
+		adEnginePhases.partners = documentLoadedPromise(
+			context.get('options.phases.partnersTimeout') || DEFAULT_PARTNERS_PHASE_WAIT_TIMEOUT,
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:28]:
```suggestion
			context.get('options.phases.partnersTimeout') ?? DEFAULT_PARTNERS_PHASE_WAIT_TIMEOUT,
```

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+			context.get('options.phases.partnersDelay') || DEFAULT_PARTNERS_PHASE_DELAY,
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:29]:
```suggestion
			context.get('options.phases.partnersDelay') ?? DEFAULT_PARTNERS_PHASE_DELAY,
```

File: src/platforms/shared/setup/ad-engine-phases-setup.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+		);
+
+		adEnginePhases.firstAdCall = waitForEventPromise(
+			eventsRepository.AD_ENGINE_SLOT_LOADED,
+			context.get('options.phases.firstAdCallTimeout') || DEFAULT_AD_CALL_WAIT_TIMEOUT,
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/ad-engine-phases-setup.ts:34]:
```suggestion
			context.get('options.phases.firstAdCallTimeout') ?? DEFAULT_AD_CALL_WAIT_TIMEOUT,
```

File: src/platforms/shared/setup/post-ad-stack-partners.setup.ts [hunk 9]
```diff
@@ -1,8 +1,6 @@
 import {
-	AdSlotEvent,
 	BrandMetrics,
 	Captify,
-	communicationService,
```
Comment (jbienkowski-fandom) [inline src/platforms/shared/setup/post-ad-stack-partners.setup.ts:5]: ATS partner is missing (ATS was part of UcpNoAds, but is not being added back here), beware that it is part of `UcpAdsMode` on ucp-desktop and ucp-mobile

File: src/platforms/shared/bidders/bidders-state.setup.ts (hunk 1 above)
Comment (mfarul) [reply src/platforms/shared/bidders/bidders-state.setup.ts:69]: I don't want to have any delays by default. If we need to do it, let's use the variable for that.

File: src/platforms/shared/setup/post-ad-stack-partners.setup.ts (hunk 9 above)
Comment (mfarul) [reply src/platforms/shared/setup/post-ad-stack-partners.setup.ts:5]: Actually do we need ATS at all here? If this is prebid enhancement we don't call it. We might send the hashes but will never be used if user selects noads.

**Review by jbienkowski-fandom — APPROVED** [review]

**Review by preludek — DISMISSED** [review]
WIth @adi-wesolowski-fandom's blessing 🙏

**Review by preludek — APPROVED** [review]

**Pull request merged**